src/mixins/filing-template-mixin.ts

```diff
@@ -352,43 +352,57 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
     }
 
-    // Create parties list with completing party
-    // (assumes CP doesn't already exist in array)
-    let parties = cloneDeep(this.getPeopleAndRoles) // make a copy
-    parties.push({
-      officer: {
-        partyType: PartyTypes.PERSON,
-        firstName: this.getCompletingParty.firstName,
-        middleName: this.getCompletingParty.middleName,
-        lastName: this.getCompletingParty.lastName
-      },
-      roles: [{ roleType: RoleTypes.COMPLETING_PARTY }],
-      mailingAddress: this.getCompletingParty.mailingAddress
-    } as OrgPersonIF)
-
-    // Add changed parties
-    if (this.hasPeopleAndRolesChanged) {
-      // if filing and paying, filter out removed entities and omit the 'actions' properties
+    // Apply parties to filing
+    {
+      let parties = cloneDeep(this.getPeopleAndRoles) // make a copy
+
+      // Add completing party
+      // (assumes CP doesn't already exist in array)
+      parties.push({
+        officer: {
+          partyType: PartyTypes.PERSON,
+          firstName: this.getCompletingParty.firstName,
+          middleName: this.getCompletingParty.middleName,
+          lastName: this.getCompletingParty.lastName
+        },
+        roles: [{
+          roleType: RoleTypes.COMPLETING_PARTY,
+          appointmentDate: this.getCurrentDate
+        }],
+        mailingAddress: this.getCompletingParty.mailingAddress
+      } as OrgPersonIF)
+
+      // if filing...
       if (!isDraft) {
-        // Filter out parties actions
+        // Filter out removed parties and delete "actions" property
         parties = parties.filter(x => !x.actions?.includes(ActionTypes.REMOVED))
-          .map((x) => {
-            // Remove null or undefined properties to adhere to schema.
-            // ie taxId is an optional property but schema doesn't accept defined empty properties
-            Object.keys(x.officer).forEach(key => {
-              if (x.officer[key] === null || x.officer[key] === '') {
-                delete x.officer[key]
-              }
-            })
-
-            const { actions, ...rest } = x
-            return rest
-          })
+          .map((x) => { const { actions, ...rest } = x; return rest })
       }
-    }
 
-    // Apply parties to filing
-    filing.changeOfRegistration.parties = parties
+      // fix schema issues
+      parties = parties.map(party => {
+        // remove empty Officer properties -- schema doesn't accept None (null) properties
+        Object.keys(party.officer).forEach(key => {
+          if (!party.officer[key]) delete party.officer[key]
+        })
+
+        // remove empty Delivery Address and Mailing Address properties
+        if (party.deliveryAddress) {
+          if (!party.deliveryAddress.streetAddressAdditional) delete party.deliveryAddress.streetAddressAdditional
+          if (!party.deliveryAddress.deliveryInstructions) delete party.deliveryAddress.deliveryInstructions
```

Comment:
These are optional fields but we cannot set them as `null` since the API will see them as `None`, which is not a string type and therefore throws a HTTP 422 response.